services/ui_backend_service/data/cache/get_task_action.py

```diff
@@ -0,0 +1,21 @@
+from typing import List
+
+from .get_data_action import GetData
+
+from metaflow import Task
+
+
+class GetTask(GetData):
+    @classmethod
+    def format_request(cls, pathspecs: List[str], invalidate_cache=False):
+        return super().format_request(targets=pathspecs, invalidate_cache=invalidate_cache)
+
+    @classmethod
+    def fetch_data(cls, pathspec):
+        task = Task(pathspec)
```

Comment:
still todo: implement error wrapping for the data fetching, and streaming it out to the action as previously.

the rewrite looks excellent though, very clean! :)

Reply:
I'm implementing a decorator for the log feature which will wrap the S3 errors raised by the metaflow library, and use the existing custom errors with corresponding id's so this is already on its way.

what seems to be needed in this case is access to the `stream_output` from `execute()` inside fetch_data to be able to stream errors.

Reply:
Added `stream_output` to fetch_data.